spec/rubocop/cop/rspec/verified_double_reference_spec.rb

```diff
@@ -0,0 +1,90 @@
+# frozen_string_literal: true
+
+RSpec.shared_examples 'verified double references' do |verified_double|
+  let(:cop_class) { RuboCop::Cop::RSpec::VerifiedDoubleReference }
+
+  context 'when EnforcedStyle is constant' do
+    let(:cop_config) do
+      { 'EnforcedStyle' => 'constant' }
+    end
+
+    it 'does not flag a violation when using a constant reference' do
+      expect_no_offenses("#{verified_double}(ClassName)")
+    end
+
+    it 'flags a violation when using a string reference' do
+      expect_offense(
+        aligned_offense(
+          "#{verified_double}('ClassName')",
+          '^^^^^^^^^^^ Prefer `constant` references for verified doubles.'
+        )
+      )
+
+      expect_correction(<<~RUBY)
+        #{verified_double}(ClassName)
+      RUBY
+    end
+
+    include_examples 'detects style',
+                     "#{verified_double}(ClassName)",
+                     'constant'
+  end
+
+  context 'when EnforcedStyle is string' do
+    let(:cop_config) do
+      { 'EnforcedStyle' => 'string' }
+    end
+
+    it 'does not flag a violation when using a string reference' do
+      expect_no_offenses("#{verified_double}('ClassName')")
+    end
+
+    it 'flags a violation when using a constant reference' do
+      expect_offense(
+        aligned_offense(
+          "#{verified_double}(ClassName)",
+          '^^^^^^^^^ Prefer `string` references for verified doubles.'
+        )
+      )
+
+      expect_correction(<<~RUBY)
+        #{verified_double}('ClassName')
+      RUBY
+    end
+
+    include_examples 'detects style',
+                     "#{verified_double}('ClassName')",
+                     'string'
+  end
+
+  it 'does not flag a violation when reference is not a supported style' do
+    expect_no_offenses(<<~RUBY)
+      klass = Array
+      instance_double(klass)
+
+
+      @sut = Array
+      let(:double) { instance_double(@sut) }
+
+      object_double([])
+
+      class_double(:Model)
+    RUBY
+  end
+end
+
+RSpec.describe RuboCop::Cop::RSpec::VerifiedDoubleReference do
+  described_class::RESTRICT_ON_SEND.each do |verified_double|
+    describe verified_double do
+      it_behaves_like 'verified double references', verified_double
+    end
+  end
+end
+
+def aligned_offense(offender, offense)
+  offense_start = offender.index('(')
+  aligned = offender + "\n#{' ' * (offense_start + 1)}#{offense}"
+  <<~RUBY
+    #{aligned}
+  RUBY
+end
```

Comment:
I understand your pain 😬 We actually have a (well-hidden) helper for these kinds of specs with variable-length indentation – it’s documented at https://github.com/rubocop/rubocop/blob/5b6e3a2d8751a434079f5ce359bf34f358d6c2ea/lib/rubocop/rspec/expect_offense.rb#L75-L102

If you like, I can change the specs to not use `#aligned_offense`. Either I can push a commit to your branch (if possible), or I can do it after merge.

Reply:
Turns out the diff isn’t so bad:
```diff
diff --git a/spec/rubocop/cop/rspec/verified_double_reference_spec.rb b/spec/rubocop/cop/rspec/verified_double_reference_spec.rb
index 930a9145..a90a8e8d 100644
--- a/spec/rubocop/cop/rspec/verified_double_reference_spec.rb
+++ b/spec/rubocop/cop/rspec/verified_double_reference_spec.rb
@@ -13,12 +13,10 @@ RSpec.shared_examples 'verified double references' do |verified_double|
     end
 
     it 'flags a violation when using a string reference' do
-      expect_offense(
-        aligned_offense(
-          "#{verified_double}('ClassName')",
-          '^^^^^^^^^^^ Prefer `constant` references for verified doubles.'
-        )
-      )
+      expect_offense(<<~RUBY, verified_double: verified_double)
+        %{verified_double}('ClassName')
+        _{verified_double} ^^^^^^^^^^^ Prefer `constant` references for verified doubles.
+      RUBY
 
       expect_correction(<<~RUBY)
         #{verified_double}(ClassName)
@@ -40,12 +38,10 @@ RSpec.shared_examples 'verified double references' do |verified_double|
     end
 
     it 'flags a violation when using a constant reference' do
-      expect_offense(
-        aligned_offense(
-          "#{verified_double}(ClassName)",
-          '^^^^^^^^^ Prefer `string` references for verified doubles.'
-        )
-      )
+      expect_offense(<<~RUBY, verified_double: verified_double)
+        %{verified_double}(ClassName)
+        _{verified_double} ^^^^^^^^^ Prefer `string` references for verified doubles.
+      RUBY
 
       expect_correction(<<~RUBY)
         #{verified_double}('ClassName')
@@ -80,11 +76,3 @@ RSpec.describe RuboCop::Cop::RSpec::VerifiedDoubleReference do
     end
   end
 end
-
-def aligned_offense(offender, offense)
-  offense_start = offender.index('(')
-  aligned = offender + "\n#{' ' * (offense_start + 1)}#{offense}"
-  <<~RUBY
-    #{aligned}
-  RUBY
-end
```
(e.g. copy to clipboard and apply with `pbpaste | git apply` if on MacOS)

Reply:
@bquorning Nice! Applied